enhancements/operator-sdk/downstream-operator-sdk.md

```diff
@@ -0,0 +1,391 @@
+  * As a Red Hat operator customer, I would want to have a released downstream Operator SDK base image to use to build my own operators.
+  * Provide operator-lib downstream
+* Create shell-script to help automate the process
+  * As an operator-sdk developer, I would like a way to easily bring upstream SDK releases downstream for release with OpenShift
+* Continue delivering Ansible and Helm operator base images
+  * As an Ansible operator developer, I would want to continue having a released downstream Ansible operator image I could base my operator on.
+  * As a Helm operator developer, I would want to continue having  a released downstream Helm operator image I could base my operator on.
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Repos
+
+Mirror the upstream Operator SDK repos downstream in the openshift GitHub
+organization. We will have a one to one upstream to downstream repo.
+
+* operator-sdk - github.com/openshift/operator-sdk
+* operator-lib - github.com/openshift/operator-lib
+* ocp-release-operator-sdk - to be retired after the above repos are working
+
+The downstream repo will contain a special _overlay_ branch that contains all of
+the downstream-specific changes. The files to be contained in this overlay are
+as follows:
+
+* Dockerfiles to support CI
+* Dockerfiles to build downstream images
+* Vendor Go dependencies
+* Ansible collections to support Ansible operator
+
+We will pin the `downstream:master` to `upstream:master` and
+_upstream_release_branches_ to _downstream_release_branches_. This means we will
+manage the OpenShift release branches ourselves. So `release-x.y` will track a
+specific upstream branch that was mirrored. Only tagged commits will ever be
+brought into the release branches.
+
+#### Syncing
+
+Today, we sync `operator-framework/operator-sdk` into
+`openshift/ocp-release-operator-sdk`. We sync it manually with `git merge`. The
+`master` branch downstream currently tracks the latest OpenShift release. So
+when moving to a new major SDK release downstream, the `master` branch sync is
+frought with conflicts that can cause missing changes or just make for a very
+tedious merge.
+
+The idea behind the autosyncing strategy is that the downstream branches will
+match the upstream branches 1 for 1. So downstream `master` will track upstream
+`master`. Downstream `vx.y.z` will track upstream `vx.y.z`. So the syncs should
+occur with minimal conflicts. These downstream branches will NOT contain any
+special downstream fixes, they are strictly mirrors of the upstream.
+
+We will manage our own OpenShift release branches. These branches will be a
+combination of a tagged upstream branch plus the overlay branch.
+
+There is an initial prototype of the syncing code already:
+
+https://github.com/fabianvf/ocp-osdk-downstream-automation
+
+#### Overlay Branch Directory Structures
+
+Today, we have the following files in the `ocp-release-operator-sdk` repo. This
```

Comment:
Just a doubt. What is the difference between the tests present in `ocp-release-operator-sdk/ci` similar to `ocp-release-operator-sdk/hack/tests/`, as both scaffold an example operator and run it on Kind cluster. If both are similar would we be removing hack/tests from overlay branch, or tests in ci/ from upstream branch.

Reply:
@varshaprasad96 I've removed the `tests` directory from the overlay. I think we need to run the upstream tests downstream in OpenShift. I've added a user story to cover that. It is still unknown what will need to change but this will be done in a jira story in the epic to figure out what needs to happen to allow them to run.